JavaFXUserInterface/src/main/java/Utilities/UIUtilities.java

```diff
@@ -17,12 +20,49 @@
 
 public class UIUtilities {
 
+    /**
+     * Given a tableView this function will set the width to fit the largest content
+     *
+     * @author Paul
+     */
+    public static void autoResizeColumns(TableView<?> table) {
+        table.setColumnResizePolicy(TableView.UNCONSTRAINED_RESIZE_POLICY);
+        table.getColumns().forEach(column ->
+        {
+            Text t = new Text(column.getText());
+            double max = t.getLayoutBounds().getWidth();
+            for (int i = 0; i < table.getItems().size(); i++) {
+                if (column.getCellData(i) != null) {
+                    t = new Text(column.getCellData(i).toString());
+                    double calcWidth = t.getLayoutBounds().getWidth();
+                    if (calcWidth > max) {
+                        max = calcWidth;
+                    }
+                }
+            }
+            column.setPrefWidth(max + 35.0d);
+        });
+    }
+
+    /**
+     * This function validates an input of a change on a text field to only allow the change if it fits the DecimalFormat
+     *
+     * @author Paul
+     */
+    public static TextFormatter.Change checkFormat(DecimalFormat format, TextFormatter.Change c) {
+        if (c.getControlNewText().isEmpty())
+            return c;
+        ParsePosition parsePosition = new ParsePosition(0);
+        if (format.parse(c.getControlNewText(), parsePosition) == null || parsePosition.getIndex() < c.getControlNewText().length())
+            return null;
+        return c;
+    }
+
```

Comment:
why move this function a 100 lines up...

Reply:
I didn't manually change it. I am guessing it did automatically when I did the code cleanup by IntelliJ. It automatically moves static functions at the top of the file.